posthog/api/insight.py

```diff
@@ -136,21 +137,23 @@ def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Insight:
         request = self.context["request"]
         team = Team.objects.get(id=self.context["team_id"])
         validated_data.pop("last_refresh", None)  # last_refresh sometimes gets sent if dashboard_item is duplicated
+        tags = validated_data.pop("tags", None)  # tags are created separately as global tag relationships
 
         if not validated_data.get("dashboard", None) and not validated_data.get("dive_dashboard", None):
             dashboard_item = Insight.objects.create(
                 team=team, created_by=request.user, last_modified_by=request.user, **validated_data
             )
-            return dashboard_item
         elif validated_data["dashboard"].team == team:
             created_by = validated_data.pop("created_by", request.user)
             dashboard_item = Insight.objects.create(
                 team=team, last_refresh=now(), created_by=created_by, last_modified_by=created_by, **validated_data
             )
-            return dashboard_item
         else:
             raise serializers.ValidationError("Dashboard not found")
 
+        self._attempt_set_tags(tags, dashboard_item)
```

Comment:
```suggestion
        # Manual tag creation since this create method doesn't call super()
        self._attempt_set_tags(tags, dashboard_item)
```

Good to add this here as well.